src/custom/state/claim/hooks/index.ts

```diff
@@ -1000,3 +1000,70 @@ export const useClaimLinks = () => {
     [chainId]
   )
 }
+
+/**
+ * Hook that parses the result input with BigNumber value to CurrencyAmount
+ */
+function useParseVCowResult(result: Result | undefined) {
+  const { chainId } = useActiveWeb3React()
+
+  const vCowToken = chainId ? V_COW[chainId] : undefined
+
+  return useMemo(() => {
+    if (!chainId || !vCowToken || !result) {
+      return
+    }
+
+    return CurrencyAmount.fromRawAmount(vCowToken, result[0].toString())
+  }, [chainId, result, vCowToken])
+}
+
+/**
+ * Hook that fetches the needed vCow data and returns it in VCowData type
+ */
+type VCowData = {
+  isLoading: boolean
+  total: CurrencyAmount<Currency> | undefined | null
+  unvested: CurrencyAmount<Currency> | undefined | null
+  vested: CurrencyAmount<Currency> | undefined | null
+}
+
+export function useVCowData(): VCowData {
+  const vCowContract = useVCowContract()
+  const { account } = useActiveWeb3React()
+
+  const singleCallInput = useMemo(() => [account ?? undefined], [account])
+
+  const { loading: isVestedLoading, result: vestedResult } = useSingleCallResult(
```

Comment:
i thought we were doing useSingleContractMultipleCalls or sth

Reply:
Not really a thing, my mistake. See https://github.com/gnosis/cowswap/pull/2552#discussion_r830156098